packages/odata-service-writer/src/index.ts

```diff
@@ -39,61 +66,55 @@ async function generate(basePath: string, service: OdataService, fs?: Editor): P
     if (!fs) {
         fs = create(createStorage());
     }
-    validateBasePath(basePath, fs);
+    const paths = await findProjectFiles(basePath, fs);
+    validateBasePath(basePath, ['webapp/manifest.json'], fs);
     enhanceData(service);
 
     // merge content into existing files
-    const templateRoot = join(__dirname, '..', 'templates');
-    const extRoot = join(templateRoot, 'extend');
+    const templateRoot = join(__dirname, '../templates');
 
     // manifest.json
-    const manifestPath = join(basePath, 'webapp', 'manifest.json');
-    // Get component app id
-    const manifest = fs.readJSON(manifestPath);
-    const appProp = 'sap.app';
-    const appid = manifest?.[appProp]?.id;
-    // Throw if required property is not found manifest.json
-    if (!appid) {
-        throw new Error(
-            t('error.requiredProjectPropertyNotFound', { property: `'${appProp}'.id`, path: manifestPath })
-        );
-    }
+    updateManifest(basePath, service, fs, templateRoot);
 
-    const manifestJsonExt = fs.read(join(extRoot, `manifest.json`));
-    fs.extendJSON(manifestPath, JSON.parse(render(manifestJsonExt, service)));
+    // update ui5.yaml if it exists
+    let ui5Config: UI5Config | undefined;
+    let ui5LocalConfig: UI5Config | undefined;
+    let ui5LocalConfigPath: string | undefined;
+    if (paths.ui5Yaml) {
+        ui5Config = await UI5Config.newInstance(fs.read(paths.ui5Yaml));
+        try {
+            ui5Config.addBackendToFioriToolsProxydMiddleware(service.previewSettings as ProxyBackend);
+        } catch (error: any) {
+            if (error.message === 'error.nodeNotFound') {
+                ui5Config.addFioriToolsProxydMiddleware({ backend: [service.previewSettings as ProxyBackend] });
```

Comment:
Just confirming that this adds the backend config too

Reply:
Confirmed. `error.nodeNotFound` is thrown if the `fiori-tools-proxy` middleware is not found, so we are adding it including the proxy configuration.